src/runtime/HalideBuffer.h

```diff
@@ -617,7 +617,11 @@ class Buffer {
      * the buffer. Does nothing if this buffer did not allocate its
      * own memory. */
     void deallocate() {
-        decref();
+        if (manages_memory()) {
```

Comment:
OK, fair enough -- you recently changed it to be otherwise. Was that intentional?